src/classes/RD2_EntryFormController_TEST.cls

```diff
@@ -365,6 +366,38 @@ private with sharing class RD2_EntryFormController_TEST {
         System.assertEquals(System.Label.RD2_ErrorRDIdIsMissingForCommitment, actualException.getMessage(),
             'The exception should be match');
     }
+
+    /**
+    * @description Verifies an error record will be created when an error response is received from the Elevate API.
+    */
+    @isTest
+    private static void shouldCreateErrorRecordWhenElevateReturnsErrorResponse() {
+        RD2_EnablementService_TEST.setRecurringDonations2Enabled();
+        PS_IntegrationService.setConfiguration(PS_IntegrationServiceConfig_TEST.testConfig);
+
+        npe03__Recurring_Donation__c rd = TEST_RecurringDonationBuilder.constructEnhancedBuilder()
+            .withDefaultValues()
+            .withContact(getContact().Id)
+            .withAmount(100)
+            .withDateEstablished(START_DATE)
+            .withStartDate(START_DATE)
+            .withDayOfMonth('1')
+            .withCommitmentId(TEMP_COMMITMENT_ID)
+            .build();
+        insert rd;
+
+        Test.startTest();
+        UTIL_Http_TEST.mockBadRequestCalloutResponse();
+
+        String jsonResponse = RD2_EntryFormController.createCommitment(rd.Id, PAYMENT_METHOD_TOKEN);
+        Test.stopTest();
+
+        List<Error__c> errors = errorGateway.getRecords();
```

Comment:
Let's also verify the `Error__c.Related_Record_Id__c` equals to RD Id and `Context_Type__c` is "Elevate".